### PR TITLE
rgwlc:  additional lc notify cleanups

### PR DESCRIPTION
**Pull request by mattbenjamin**
Follow the same formula to build up obj_state and version_id in all call sites.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by dang — APPROVED** [review]

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -552,7 +549,7 @@ static int remove_expired_obj(
   del_op->params.unmod_since = meta.mtime;
 
   // notification supported only for RADOS driver for now
-  notify = driver->get_notification(dpp, oc.obj.get(), nullptr, event_type,
+  notify = driver->get_notification(dpp, obj.get(), nullptr, event_type,
```
Comment (yuvalif) [inline src/rgw/rgw_lc.cc:552]: how is the change from `oc.obj` to `obj` related to getting the object state?

Comment (mattbenjamin) [reply src/rgw/rgw_lc.cc:552]: in this commit, I'm just aligning them;  we want to get object state and generate the notification using the same object handle

**Comment by cbodley** [comment]
@mattbenjamin make check failed to build here; make sure this gets tagged for qa when it's fixed

**Comment by mattbenjamin** [comment]
ack

**Comment by mattbenjamin** [comment]
https://pulpito.ceph.com/mbenjamin-2023-11-16_20:02:51-rgw-wip-lcnotify-cleanup-distro-default-smithi/

**Comment by mattbenjamin** [comment]
https://pulpito.ceph.com/mbenjamin-2023-11-16_20:02:51-rgw-wip-lcnotify-cleanup-distro-default-smithi/
(this is against the last version, which differs only in having an unused variable that broke make check)

**Pull request merged**